roles/pulp/tasks/install.yml

```diff
@@ -167,27 +167,54 @@
         dest: "{{ pulp_install_dir }}/requirements.in"
       when: pulp_source_dir is undefined
 
-    - name: Install pip-tools before running pip-comile to check version compatibility
+    - name: Install pip-tools before running pip-compile to check version compatibility
       pip:
         name: pip-tools
         state: present
         virtualenv: '{{ pulp_install_dir }}'
         virtualenv_command: '{{ pulp_python_interpreter }} -m venv'
 
-    - name: Run pip-compile to check pulpcore/plugin compatibility
-      command: '{{ pulp_install_dir }}/bin/pip-compile'
-      args:
-        chdir: '{{ pulp_install_dir }}'
-      register: compatibility
+    - block:
+
+        # pip-compile creates requirements.txt, but we need to create it for these
+        # use cases:
+        # 1. upgrades from pulp installs older than the pip-compile check
+        # 2. Users manually modifying their venv
+        #
+        # We need requirements.txt because it affects what happens when users
+        # specify "pulp-file" (for example) without any version or an upgrade;
+        # pip-compile behaves differently depending on whether or not pulp-file is
+        # already installed (attempt to install newest version, vs leave it be.)
+        - name: list currently installed packages for the pip-compile check
+          shell: '{{ pulp_install_dir }}/bin/pip freeze > {{ pulp_install_dir }}/requirements.txt'
```

Comment:
nice!

Reply:
Thanks! I hate to use the `shell` module, but it's the simplest way to do it.